src/main/java/net/sf/jabref/model/entry/BibEntry.java

```diff
@@ -611,4 +612,21 @@ public void addKeywords(List<String> keywords) {
         // TODO Auto-generated method stub
         return fields;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
```

Comment:
Yep, this way is the right approach. However I would prefer this template (recommend by java Expert Michael Inden in his book "Der Weg zum Java Profi")
````java
@Override
public boolean equals(Object other)
        {
            if (other == null) // Null-Akzeptanz
                return false;

            if (this == other) // Reflexivität 
                return true;

            if (this.getClass() != other.getClass()) // Typgleichheit sicherstellen 
                return false;
final BibEntry = (BibEntry) other;
return .....
````
@koppor  For more info see here: https://www.artima.com/intv/bloch17.html
Equals with instanceof would additionally violate symmetry.